apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -298,13 +317,43 @@ const RegisterMetersReadingsService = new GQLCustomSchema('RegisterMetersReading
 
                 const resultRows = []
 
+                const meterNumbers = uniq(readings.map(reading => reading.meterNumber.trim()))
+                const meters = await find('Meter', {
+                    organization,
+                    number_in: meterNumbers,
+                    deletedAt: null,
+                })
+
+                const readingsWithValidDates = readings.filter(reading => isDateValid(reading.date))
+                const plainMeterReadings = await find('MeterReading', {
+                    meter: { id_in: meters.map(meter => meter.id) },
+                    date_in: uniq(readingsWithValidDates.map(reading => toISO(reading.date))),
+                })
+                const meterReadings = plainMeterReadings.map(reading => {
+                    const meter = meters.find(meter => meter.id === reading.meter)
```

Comment:
`find` has O(N) complexity, and plain is limited of `500`, so its `500` x `25000` operations. 

Can be simplified by converting `Array<Meter>` to `Record<id, Meter>`. So it's `25000` for conversion and `500` for map. 25500 total cost